lib/multi_mail/mailgun/message.rb

```diff
@@ -8,66 +8,75 @@ class Mailgun < MultiMail::Message::Base
       def mailgun_headers
-        hash = Multimap.new
+        mm = Multimap.new
         header_fields.each do |field|
           key = field.name.downcase
-          unless %w(from to cc bcc subject tag).include?(key)
-            hash["h:#{field.name}"] = field.value
+          unless %w(from to cc bcc subject tag reply-to).include?(key)
+            mm["h:#{field.name}"] = field.value
           end
         end
-        hash
+        mm.instance_variable_get('@hash')['h:Reply-To'] = self['Reply-To'].value if self['Reply-To']
+        mm
       end
 
       # Returns the message's attachments in Mailgun format.
       #
       # @return [Multimap] the attachments in Mailgun format
       # @see http://documentation.mailgun.com/user_manual.html#inline-image
       def mailgun_attachments
-        hash = Multimap.new
+        mm = Multimap.new
         attachments.each do |attachment|
           key = attachment.content_type.start_with?('image/') ? 'inline' : 'attachment'
-          hash[key] = Faraday::UploadIO.new(StringIO.new(attachment.body.decoded), attachment.content_type, attachment.filename)
+          mm[key] = Faraday::UploadIO.new(StringIO.new(attachment.body.decoded), attachment.content_type, attachment.filename)
         end
-        hash
+        mm
       end
 
       # Returns the message as parameters to POST to Mailgun.
       #
       # @return [Hash] the message as parameters to POST to Mailgun
       # @see http://documentation.mailgun.com/user_manual.html#tagging
       def to_mailgun_hash
-        hash = Multimap.new
+        mm = Multimap.new
+        hash = Hash.new
 
         %w(from subject).each do |field|
           if self[field]
-            hash[field] = self[field].value
+            mm[field] = self[field].value
           end
         end
 
         %w(to cc bcc).each do |field|
           if self[field]
             if self[field].value.respond_to?(:each)
               self[field].value.each do |value|
-                hash[field] = value
+                mm[field] = value
               end
             else
-              hash[field] = self[field].value
+              mm[field] = self[field].value
             end
           end
         end
 
+        # there may be others we want to do this with
+        %w(h:Reply-To).each do |field|
+          if self[field]
+            hash[field] = self[field].value
+          end
+        end
+
         if body_text && !body_text.empty?
-          hash['text'] = body_text
+          mm['text'] = body_text
         end
         if body_html && !body_html.empty?
-          hash['html'] = body_html
+          mm['html'] = body_html
         end
 
         tags.each do |tag|
-          hash['o:tag'] = tag
+          mm['o:tag'] = tag
         end
 
-        normalize(hash.merge(mailgun_attachments).merge(mailgun_headers).to_hash)
+        normalize(mm.to_hash.merge(mailgun_attachments).merge(mailgun_headers).merge(hash))
```

Comment:
Multimap::merge only handles merging other Multimaps and 'hash' is now actually a Hash. Hash::merge can handle Multimaps, but not the other way around.